src/main/java/org/jenkinsci/plugins/pubsub/Message.java

```diff
@@ -205,6 +231,44 @@ public T setEventName(Enum name) {
     }
 
     /**
+     * Get the Jenkins instance URL of the master from which the event was published.
+     *
+     * @return The Jenkins instance identity.
+     * @see #getJenkinsInstanceId()
+     */
+    public String getJenkinsInstanceUrl() {
```

Comment:
I am thinking if we should let enrichers provide their own instance url, for example, returning http://localhost:8081 vs https://myjenkins1.example.com? If for example I am running jenkins cluster behind firewall and based on DNS info my enricher could return whatever value it thinks should be for instance url.